raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -219,21 +228,38 @@ def handle_secretrequest(state, state_change):
     elif invalid_secretrequest:
         iteration = cancel_current_route(state)
 
+    else:
+        iteration = TransitionResult(state, list())
+
     return iteration
 
 
 def handle_secretreveal(state, state_change):
+    """ Send a balance proof to the next hop with the current mediated transfer
+    lock removed and the balance updated.
+
+    Wait for the node to notify it knows the secret and then send a new merkle
```

Comment:
what is the notion of 
> Wait for

here? Did we already wait before? Is anything below going to "wait"? Please clarify the docstring.